__tests__/integration/multisig.test.js

```diff
@@ -300,4 +321,453 @@ describe('send tx (HTR)', () => {
+    expect(response.body.success).toBe(true);
+
+    let { txHex } = response.body;
+    const txHexToDecode = txHex;
+
+    // collect signatures from 3 wallets
+    const wallets = [wallet1, wallet2, wallet3, wallet4, wallet5];
+    let signatures = await TestUtils.getXSignatures(txHex, wallets, 3);
+
+    // try to send
+    response = await TestUtils.request
+      .post('/wallet/p2sh/tx-proposal/sign-and-push')
+      .send({ txHex, signatures })
+      .set({ 'x-wallet-id': wallet1.walletId });
+
+    await TestUtils.pauseForWsUpdate();
+
+    expect(response.status).toBe(200);
+    expect(response.body.success).toBe(true);
+    expect(response.body.hash).toBeDefined();
+
+    const tokenUid = response.body.hash;
+    let mctBalance = await wallet1.getBalance(tokenUid);
+    expect(mctBalance.available).toBe(100);
+
+    // # Mint
+    response = await TestUtils.request
+      .post('/wallet/p2sh/tx-proposal/mint-tokens')
+      .send({
+        token: tokenUid,
+        amount: 1,
+        address,
+      })
+      .set({ 'x-wallet-id': wallet1.walletId });
+    expect(response.body.success).toBe(true);
+
+    txHex = response.body.txHex;
+    // collect signatures from 3 wallets
+    signatures = await TestUtils.getXSignatures(txHex, wallets, 3);
+
+    // try to send
+    response = await TestUtils.request
+      .post('/wallet/p2sh/tx-proposal/sign-and-push')
+      .send({ txHex, signatures })
+      .set({ 'x-wallet-id': wallet1.walletId });
+
+    await TestUtils.pauseForWsUpdate();
+
+    mctBalance = await wallet1.getBalance(tokenUid);
+    expect(mctBalance.available).toBe(101);
+
+    // # Melt
+    response = await TestUtils.request
+      .post('/wallet/p2sh/tx-proposal/melt-tokens')
+      .send({
+        token: tokenUid,
+        amount: 1,
+        deposit_address: address,
+        change_address: address,
+      })
+      .set({ 'x-wallet-id': wallet1.walletId });
+    expect(response.body.success).toBe(true);
+
+    txHex = response.body.txHex;
+    // collect signatures from 3 wallets
+    signatures = await TestUtils.getXSignatures(txHex, wallets, 3);
+
+    // try to send
+    response = await TestUtils.request
+      .post('/wallet/p2sh/tx-proposal/sign-and-push')
+      .send({ txHex, signatures })
+      .set({ 'x-wallet-id': wallet1.walletId });
+
+    await TestUtils.pauseForWsUpdate();
+
+    mctBalance = await wallet1.getBalance(tokenUid);
+    expect(mctBalance.available).toBe(100);
+
+    // Decode
+    response = await TestUtils.request
+      .post('/wallet/decode')
+      .send({ txHex: txHexToDecode })
+      .set({ 'x-wallet-id': wallet1.walletId });
+    expect(response.status).toBe(200);
+    expect(response.body).toEqual({
+      success: true,
+      tx: expect.objectContaining({
+        completeSignatures: false,
+        tokens: [],
+        inputs: [
+          expect.objectContaining({
+            signed: false,
+          }),
+        ],
+        outputs: expect.arrayContaining([
```

Comment:
Ok.

Reply:
Oh, it is because it can't control which input and its value in the integration test, it can be anything enough to cover the expenditures. But I will let a comment about it.